### ISSUE TITLE
Open grok doesn't support markdown tables (for README.md files for instance)

### ISSUE THREAD

**Issue by lupuleasa-core** [oracle/opengrok#2307]
Hi wonderful people,

I have worked on the docs for my enterprise codebase on github and I noticed that opengrok doesn't support markdown tables, and I was wondering if it is possible to add support for them, if needed I could contribute to it.

[Table Markdown](https://help.github.com/articles/organizing-information-with-tables/) of the form:

    | First Header  | Second Header |
    | ------------- | ------------- |
    | Content Cell  | Content Cell  |
    | Content Cell  | Content Cell  |

## Screenshots of issue:

- **How it looks on github:**

![image](https://user-images.githubusercontent.com/5671120/44205736-0fdeeb80-a160-11e8-9154-ab34d442299e.png)

- **How it looks on our opengrok:**

![image](https://i.imgur.com/I9BkTq0.png)

&nbsp;

I searched previously for "markdown" and "README.md" and I haven't seen this mentioned.

Thanks,

Lup.

**Comment by vladak** [comment]
Feel free to submit a pull request.

**Comment by lupuleasa-core** [comment]
I have looked a bit into the code, and this is what I found so far:

- From the ["markdown" search](https://github.com/oracle/opengrok/search?q=markdown&unscoped_q=markdown), it seems there is a `function init_markdown_converter() ` that utilizes a script called showdown `$.script.loadScript('js/showdown-1.4.2.min.js').done(function () {`

After googling, it seems to be related to [this](https://github.com/showdownjs/showdown) repository

The latest version is 1.8.2

Is it possible to look into upgrading this?

I can do it later on today.

**Comment by lupuleasa-core** [comment]
Just tested it on [http://demo.showdownjs.com](http://demo.showdownjs.com)

It seems tables are supported in the latest version, and I quote:

## Tables

Tables aren't part of the core Markdown spec, but they are part of GFM and Showdown supports them by turning on the option `tables`.

Colons can be used to align columns.

In the new version, the outer pipes (`|`) are optional, matching GFM spec. 

You also don't need to make the raw Markdown line up prettily.

You can also use other markdown syntax inside them.

```md
| Tables        | Are           | Cool  |
| ------------- |:-------------:| -----:|
| **col 3 is**  | right-aligned | $1600 |
| col 2 is      | *centered*    |   $12 |
| zebra stripes | ~~are neat~~  |    $1 |
```

**Comment by lupuleasa-core** [comment]
Maybe the `tables` option is not enabled.

I will look into it, but an update seems to be due.

**Comment by lupuleasa-core** [comment]
In conclusion, opengrok needs a showdown upgrade.

The 1.4.2 version is from 2016-06-21

While the table .md to .html convertor code seems to be created in 2017-12-16 - [Link](https://github.com/showdownjs/showdown/blob/3db9200d2c293d3977948558b2a4aace3e86d6ac/src/subParsers/makehtml/tables.js)

I will create a pull shortly.

**Comment by lupuleasa-core** [comment]
found the link of the newer version

https://cdn.rawgit.com/showdownjs/showdown/1.8.4/dist/showdown.min.js

**Comment by lupuleasa-core** [comment]
the change is literally a file replacement, an 1.4.2, changed to 1.8.4 in the `$.script.loadScript('js/showdown-1.4.2.min.js').done(function () {` line

My github seems to have issues with the enterprise account logged in and with the private network, will try at home.

**Comment by lupuleasa-core** [comment]
Okay, done. I had some issues with my e-mail settings on github which prevented me from pushing, now the PR is available at:

https://github.com/oracle/opengrok/pull/2310 

Test it by looking at a RADME.md with the following table inside

    | Tables        | Are           | Cool  |
    | ------------- |:-------------:| -----:|
    | **col 3 is**  | right-aligned | $1600 |
    | col 2 is      | *centered*    |   $12 |
    | zebra stripes | ~~are neat~~  |    $1 |

**Comment by vladak** [comment]
fixed in csets f96974026a5a4bba6d2b5192cea27c55e1f847d1 and 5d882bed76764e2addd8dbf6430b8183a46d1106

**Comment by vladak** [comment]
Actually, does not seem to be working for me.

**Comment by lupuleasa-core** [comment]
I found out this explanation 

`Tables aren't part of the core Markdown spec, but they are part of GFM and Showdown supports them by turning on the option "tables".`

I will look into enabling that option in the code

**Comment by lupuleasa-core** [comment]
From [the showdown options page](https://github.com/showdownjs/showdown/wiki/Showdown-Options) it seems we have to enable `tables` like this:

    var converter = new showdown.Converter();
    converter.setOption('tables', true);

**Comment by lupuleasa-core** [comment]
#2311 Fix added, test markdown available there

**Comment by vladak** [comment]
fixed in cset 046743cd (+ follow-ups)
